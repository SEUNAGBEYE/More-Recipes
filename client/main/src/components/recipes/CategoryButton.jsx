import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';


/**
 * @description - Component for selecting category to display
 * @param {Object} props
 *
 * @returns {jsx} JSX
 */
const CategoryButton = (props) => (
  <div className="dropdown" style={{ float: 'left' }}
    data-category-button="category-button"
  >
    <button className="btn btn-default dropdown-toggle auth-button"
      type="button" id="about-us" data-toggle="dropdown"
      aria-haspopup="true" aria-expanded="false">Category
    </button>
    <div className="dropdown-menu" aria-labelledby="about-us">
      {
        props.categories.map(category =>
          (<Link className="dropdown-item"
            key={category.id} to={`/categories/${category.name}`}
            data-category-link={`category-link-${category.name}`}
          >{category.name}</Link>))
      }
    </div>
  </div>
);

const propTypes = {
  categories: PropTypes.array.isRequired,
};

CategoryButton.propTypes = propTypes;

/**
 * mapStateToProps
 * @export
 * @param {Object} state
 *
 * @returns {Object} object
 */
export const mapStateToProps = (state) => ({
  categories: state.recipes.recipeCategories
    .map(category => ({ name: category.name, id: category.id }))
});

export { CategoryButton as CategoriesButton };
export default connect(mapStateToProps)(CategoryButton);
