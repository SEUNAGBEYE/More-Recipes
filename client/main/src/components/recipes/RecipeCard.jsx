import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import DeleteModal from './DeleteModal';
import EditModal from './EditModal';
import checkAuth from '../../../utils/CheckAuth';
import RecipeCardActions from '../recipes/RecipeCardActions';

import {
  toggleThumbsDownRecipe,
  toggleThumbsUpRecipe,
  getFavouritedRecipesIds,
  toggleFavouriteRecipe,
  deleteRecipe,
  editRecipe
}
  from '../../actions/Recipes';


/**
 * @export
 * @class RecipeCard
 * @extends {Component}
 */
export class RecipeCard extends Component {
  /**
   * Creates an instance of RecipeCard.
   * @param {Object} props
   * @memberof RecipeCard
   */
  constructor(props) {
    super(props);
    this.toggleFavouriteRecipe = this.toggleFavouriteRecipe.bind(this);
    this.toggleThumbsDownRecipe = this.toggleThumbsDownRecipe.bind(this);
    this.toggleThumbsUpRecipe = this.toggleThumbsUpRecipe.bind(this);
    this.deleteRecipe = this.deleteRecipe.bind(this);
    this.editRecipe = this.editRecipe.bind(this);

    this.state = {
      recipe: {
        name: '',
        description: '',
        upvotes: [],
        downvotes: []
      },
      isDownVoted: '',
      isUpVoted: ''
    };
  }

  /**
   * @returns {void} void
   * @memberof RecipeCard
   */
  componentWillMount() {
    const { recipe, user } = this.props;
    this.setState({
      recipe,
      isUpVoted: recipe.upvotes.includes(parseInt(user.userId, 10)),
      isDownVoted: recipe.downvotes.includes(parseInt(user.userId, 10))
    });
  }

  /**
   * @param {Object} nextProps
   *
   * @returns {void} void
   * @memberof RecipeCard
   */
  componentWillReceiveProps(nextProps) {
    if (this.props.recipe !== nextProps.recipe) {
      const { recipe } = nextProps;
      const { user } = this.props;
      this.setState({
        recipe,
        isUpVoted: recipe.upvotes.includes(parseInt(user.userId, 10)),
        isDownVoted: recipe.downvotes.includes(parseInt(user.userId, 10))
      });
    }
  }

  /**
   * @param {Object} event
   *
   * @returns {void} void
   * @memberof RecipeCard
   */
  toggleFavouriteRecipe(event) {
    event.preventDefault();
    const { isAuthenticated, history } = this.props;
    if (isAuthenticated) {
      this.props.toggleFavouriteRecipe(parseInt(event.target.id, 10));
    }
    checkAuth(isAuthenticated, history);
  }

  /**
   * @param {any} event
   *
   * @returns {void} void
   * @memberof RecipeCard
   */
  toggleThumbsUpRecipe(event) {
    event.preventDefault();
    if (this.props.isAuthenticated) {
      this.props.toggleThumbsUpRecipe(event.target.id);
      this.setState({ toggleThumbsUp: !this.state.toggleThumbsUp });
    }
    checkAuth(this.props.isAuthenticated, this.props.history);
  }

  /**
   * @param {Object} event
   *
   * @returns {void} void
   * @memberof RecipeCard
   */
  deleteRecipe(event) {
    const { id } = event.target;
    this.props.deleteRecipe(id);
  }

  /**
   * @param {Number} id
   * @param {Object} recipe
   *
   * @returns {void} void
   * @memberof RecipeCard
   */
  editRecipe(id, recipe) {
    return this.props.editRecipe(id, recipe);
  }

  /**
   * @param {Object} event
   *
   * @returns {Object} object
   * @memberof RecipeCard
   */
  toggleThumbsDownRecipe(event) {
    event.preventDefault();
    if (this.props.isAuthenticated) {
      this.props.toggleThumbsDownRecipe(event.target.id);
      this.setState({ toggleThumbsUp: !this.state.toggleThumbsUp });
    }
    checkAuth(this.props.isAuthenticated, this.props.history);
  }


  /**
   * @returns {JSX} jsx
   * @memberof RecipeCard
   */
  render() {
    const { recipe, myFavouriteRecipes } = this.props;
    const isFavorited = myFavouriteRecipes.includes(parseInt(recipe.id, 10));
    return (
      <div className="col-xs-12 col-sm-12 col-md-6 col-lg-3 my-card">
        <div className="card recipe-card">
          <img className="card-img-top"
            src={this.state.recipe.image}
            alt="Card image cap"
          />
          <div className="container">
            <div className="card-block">
              <Link to={`/recipe/${this.props.id}`}
                data-recipe-card-id={`recipe${this.props.id}`}
              >
                <h4 className="card-title break-word">
                  {this.state.recipe.name.slice(0, 15)}
                </h4>
                <p className="card-text break-word">
                  {this.state.recipe.description.slice(0, 80)}
                </p>
              </Link>
              <RecipeCardActions
                isDownVoted={this.state.isDownVoted}
                isUpVoted={this.state.isUpVoted}
                isFavorited={isFavorited}
                user={this.props.user}
                recipe={this.props.recipe}
                toggleFavouriteRecipe={this.toggleFavouriteRecipe}
                toggleThumbsDownRecipe={this.toggleThumbsDownRecipe}
                toggleThumbsUpRecipe={this.toggleThumbsUpRecipe}/>
            </div>
          </div>
        </div>
        <EditModal recipe={this.props.recipe}
          editRecipe={this.editRecipe}
          id={this.props.recipe.id}
        />
        <DeleteModal id={this.props.id} onDelete={this.deleteRecipe}/>
      </div>
    );
  }
}

const propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  user: PropTypes.object.isRequired,
  history: PropTypes.object,
  recipe: PropTypes.object.isRequired,
  myFavouriteRecipes: PropTypes.array.isRequired,
  toggleFavouriteRecipe: PropTypes.func.isRequired,
  getFavouritedRecipesIds: PropTypes.func.isRequired,
  toggleThumbsDownRecipe: PropTypes.func.isRequired,
  toggleThumbsUpRecipe: PropTypes.func.isRequired,
  deleteRecipe: PropTypes.func.isRequired,
  editRecipe: PropTypes.func.isRequired,
};

RecipeCard.propTypes = propTypes;

/**
 * mapStateToProps
 * @export
 * @param {Object} state
 *
 * @returns {object} object
 */
export const mapStateToProps = (state) => ({
  user: state.auth.user,
  isAuthenticated: state.auth.isAuthenticated,
  myFavouriteRecipes: state.auth.isAuthenticated ? state.recipes.userFavouritedRecipeId : []

});

export default connect(mapStateToProps, {
  toggleFavouriteRecipe,
  getFavouritedRecipesIds,
  toggleThumbsUpRecipe,
  toggleThumbsDownRecipe,
  deleteRecipe,
  editRecipe
})(RecipeCard);

