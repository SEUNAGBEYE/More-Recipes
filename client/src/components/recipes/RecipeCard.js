import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import {connect} from 'react-redux';
import Recipe from '../../actions/Recipes';
import { pasta, seun } from '../../helpers/Images';
import RecipeModal from './RecipeModal';
import DeleteModal from './DeleteModal';
import {editRecipe} from '../../actions/Recipes'
import EditModal from './EditModal';


/**
 * @class RecipeCard
 */
export default class RecipeCard extends Component{

  /**
   * @memberOf RecipeCard
   * return {object} object
   */
  render(){
    return (
				<div className="col-xs-12 col-sm-12 col-md-6 col-lg-3 my-card">
					<Link to={`/recipe/${this.props.id}`}>
					<div className="card">
					  	<img className="card-img-top" src={pasta} alt="Card image cap" />
						<div className="container">
							<div className="card-block">
					    		<h4 className="card-title">{this.props.recipe.name}</h4>
					    		<p className="card-text">{this.props.recipe.description || "Some quick example text to build on the card title and make up the bulk of the card's content."}</p>
							    <div className="d-flex justify-content-between">
										<Link to="/" className="fa fa-eye icons">10</Link>
										<Link to="/" className="fa fa-pencil icons" data-toggle="modal" data-target={`#editModal${this.props.id}`}></Link>
										<Link to="/" className="fa fa-trash icons" data-toggle="modal" data-target={`#deleteModal${this.props.id}`}></Link>
								 </div>
							</div>
						</div>
					</div>
					<EditModal recipe={this.props.recipe} editRecipe={this.props.editRecipe} id={this.props.id}/>
					<DeleteModal id={this.props.id} onDelete={this.props.onDelete}/>
					</Link>	
        </div>
				
    )
  }
}

// const mapStateToProps = (state) => {
// 	console.log(state)
//   return {
// 		recipes: [...state]
// 	}
// };

// export default connect(mapStateToProps, {editRecipe})(RecipeCard);

