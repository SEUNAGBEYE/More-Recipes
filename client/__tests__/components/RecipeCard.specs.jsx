
import React from 'react';
import { shallow } from 'enzyme';
import { RecipeCard, mapStateToProps } from '../../main/src/components/recipes/RecipeCard';

const state = {
  auth: {
    user: {
      firstName: 'Seun',
      lastName: 'Agbeye'
    },
    isAuthenticated: true
  },
  recipes: {
    userFavouritedRecipeId: []
  }
};

const recipe = {
  name: 'Amala',
  image: 'This is the image',
  upvotes: [1,2,4],
  downvotes: [1,2,4],
  description: 'This is the decription'
}


const props = {
  isAuthenticated: state.auth.isAuthenticated,
  recipe,
  user: {
    firstName: 'Seun',
    lastName: 'Agbeye'
  },
  myFavouriteRecipes: [recipe],
  deleteRecipe: jest.fn(),
  editRecipe: jest.fn(),
  toggleThumbsUpRecipe: jest.fn(),
  toggleThumbsDownRecipe: jest.fn(),
  getFavouritedRecipesIds: jest.fn(),
  toggleFavouriteRecipe: jest.fn(),
  pagination: 1,
  history: {
    push: jest.fn(),
    location: {
      pathname: 'recipes'
    }
  }
};

const event = {
  preventDefault: jest.fn(),
  target: {
    id: 1
  }
}

describe('# RecipeCard', () => {
  const wrapper = shallow(
      <RecipeCard {...props} />
  );

  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    expect(wrapper.getElement().type).toBe('div');
    expect(wrapper).toMatchSnapshot()
    done();
  });

  it('should call deleteRecipe', (done) => {
    const deleteRecipeSpy = jest.spyOn(wrapper.instance(), 'deleteRecipe')
    wrapper.instance().deleteRecipe(event);
    expect(deleteRecipeSpy).toHaveBeenCalledTimes(1)
    done();
  });

  it('should call editRecipe', (done) => {
    const editRecipeSpy = jest.spyOn(wrapper.instance(), 'editRecipe')
    wrapper.instance().editRecipe(event.target.id, recipe);
    expect(editRecipeSpy).toHaveBeenCalledTimes(1);
    done();
  });

  it('should call toggleThumbsUpRecipe', (done) => {
    const toggleThumbsUpRecipeSpy = jest.spyOn(wrapper.instance(), 'toggleThumbsUpRecipe')
    wrapper.instance().toggleThumbsUpRecipe(event);
    expect(toggleThumbsUpRecipeSpy).toHaveBeenCalledTimes(1)
    done();
  });

  it('should call toggleThumbsDownRecipe', (done) => {
    const toggleThumbsDownRecipeSpy = jest.spyOn(wrapper.instance(), 'toggleThumbsDownRecipe')
    wrapper.instance().toggleThumbsDownRecipe(event);
    expect(toggleThumbsDownRecipeSpy).toHaveBeenCalledTimes(1);
    done();
  });

  it('should call toggleFavouriteRecipe', (done) => {
    const toggleFavouriteRecipeSpy = jest.spyOn(wrapper.instance(), 'toggleFavouriteRecipe')
    wrapper.instance().toggleFavouriteRecipe(event);
    expect(toggleFavouriteRecipeSpy).toHaveBeenCalledTimes(1);
    done();
  });

  it('should call componentWillReceiveProps', (done) => {
    const componentWillReceivePropsSpy = jest.spyOn(wrapper.instance(), 'componentWillReceiveProps')
    wrapper.instance().componentWillReceiveProps(props);
    expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(1);
    done();
  });

  it('should call mapStateToProps(state)', (done) => {
    mapStateToProps(state);
    done();
  }); 
});
