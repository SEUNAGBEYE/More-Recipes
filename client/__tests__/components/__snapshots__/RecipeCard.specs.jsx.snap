// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# RecipeCard should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeCard
    deleteRecipe={[MockFunction]}
    editRecipe={[MockFunction]}
    getFavouritedRecipesIds={[MockFunction]}
    history={
      Object {
        "location": Object {
          "pathname": "recipes",
        },
        "push": [MockFunction],
      }
    }
    isAuthenticated={true}
    myFavouriteRecipes={
      Array [
        Object {
          "description": "This is the decription",
          "downvotes": Array [
            1,
            2,
            4,
          ],
          "image": "This is the image",
          "name": "Amala",
          "upvotes": Array [
            1,
            2,
            4,
          ],
        },
      ]
    }
    pagination={1}
    recipe={
      Object {
        "description": "This is the decription",
        "downvotes": Array [
          1,
          2,
          4,
        ],
        "image": "This is the image",
        "name": "Amala",
        "upvotes": Array [
          1,
          2,
          4,
        ],
      }
    }
    toggleFavouriteRecipe={[MockFunction]}
    toggleThumbsDownRecipe={[MockFunction]}
    toggleThumbsUpRecipe={[MockFunction]}
    user={
      Object {
        "firstName": "Seun",
        "lastName": "Agbeye",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card recipe-card"
        >
          <img
            alt="Card image cap"
            className="card-img-top"
            src="This is the image"
          />
          <div
            className="container"
          >
            <div
              className="card-block"
            >
              <Link
                data-recipe-card-id="recipeundefined"
                replace={false}
                to="/recipe/undefined"
              >
                <h4
                  className="card-title"
                >
                  Amala
                </h4>
                <p
                  className="card-text"
                >
                  This is the decription
                </p>
              </Link>
              <RecipeCardActions
                isDownVoted={false}
                isFavorited={false}
                isUpVoted={false}
                recipe={
                  Object {
                    "description": "This is the decription",
                    "downvotes": Array [
                      1,
                      2,
                      4,
                    ],
                    "image": "This is the image",
                    "name": "Amala",
                    "upvotes": Array [
                      1,
                      2,
                      4,
                    ],
                  }
                }
                toggleFavouriteRecipe={[Function]}
                toggleThumbsDownRecipe={[Function]}
                toggleThumbsUpRecipe={[Function]}
                user={
                  Object {
                    "firstName": "Seun",
                    "lastName": "Agbeye",
                  }
                }
              />
            </div>
          </div>
        </div>,
        <EditModal
          editRecipe={[Function]}
          id={undefined}
          recipe={
            Object {
              "description": "This is the decription",
              "downvotes": Array [
                1,
                2,
                4,
              ],
              "image": "This is the image",
              "name": "Amala",
              "upvotes": Array [
                1,
                2,
                4,
              ],
            }
          }
        />,
        <DeleteModal
          id={undefined}
          onDelete={[Function]}
        />,
      ],
      "className": "col-xs-12 col-sm-12 col-md-6 col-lg-3 my-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Card image cap"
              className="card-img-top"
              src="This is the image"
            />,
            <div
              className="container"
            >
              <div
                className="card-block"
              >
                <Link
                  data-recipe-card-id="recipeundefined"
                  replace={false}
                  to="/recipe/undefined"
                >
                  <h4
                    className="card-title"
                  >
                    Amala
                  </h4>
                  <p
                    className="card-text"
                  >
                    This is the decription
                  </p>
                </Link>
                <RecipeCardActions
                  isDownVoted={false}
                  isFavorited={false}
                  isUpVoted={false}
                  recipe={
                    Object {
                      "description": "This is the decription",
                      "downvotes": Array [
                        1,
                        2,
                        4,
                      ],
                      "image": "This is the image",
                      "name": "Amala",
                      "upvotes": Array [
                        1,
                        2,
                        4,
                      ],
                    }
                  }
                  toggleFavouriteRecipe={[Function]}
                  toggleThumbsDownRecipe={[Function]}
                  toggleThumbsUpRecipe={[Function]}
                  user={
                    Object {
                      "firstName": "Seun",
                      "lastName": "Agbeye",
                    }
                  }
                />
              </div>
            </div>,
          ],
          "className": "card recipe-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Card image cap",
              "className": "card-img-top",
              "src": "This is the image",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="card-block"
              >
                <Link
                  data-recipe-card-id="recipeundefined"
                  replace={false}
                  to="/recipe/undefined"
                >
                  <h4
                    className="card-title"
                  >
                    Amala
                  </h4>
                  <p
                    className="card-text"
                  >
                    This is the decription
                  </p>
                </Link>
                <RecipeCardActions
                  isDownVoted={false}
                  isFavorited={false}
                  isUpVoted={false}
                  recipe={
                    Object {
                      "description": "This is the decription",
                      "downvotes": Array [
                        1,
                        2,
                        4,
                      ],
                      "image": "This is the image",
                      "name": "Amala",
                      "upvotes": Array [
                        1,
                        2,
                        4,
                      ],
                    }
                  }
                  toggleFavouriteRecipe={[Function]}
                  toggleThumbsDownRecipe={[Function]}
                  toggleThumbsUpRecipe={[Function]}
                  user={
                    Object {
                      "firstName": "Seun",
                      "lastName": "Agbeye",
                    }
                  }
                />
              </div>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    data-recipe-card-id="recipeundefined"
                    replace={false}
                    to="/recipe/undefined"
                  >
                    <h4
                      className="card-title"
                    >
                      Amala
                    </h4>
                    <p
                      className="card-text"
                    >
                      This is the decription
                    </p>
                  </Link>,
                  <RecipeCardActions
                    isDownVoted={false}
                    isFavorited={false}
                    isUpVoted={false}
                    recipe={
                      Object {
                        "description": "This is the decription",
                        "downvotes": Array [
                          1,
                          2,
                          4,
                        ],
                        "image": "This is the image",
                        "name": "Amala",
                        "upvotes": Array [
                          1,
                          2,
                          4,
                        ],
                      }
                    }
                    toggleFavouriteRecipe={[Function]}
                    toggleThumbsDownRecipe={[Function]}
                    toggleThumbsUpRecipe={[Function]}
                    user={
                      Object {
                        "firstName": "Seun",
                        "lastName": "Agbeye",
                      }
                    }
                  />,
                ],
                "className": "card-block",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <h4
                        className="card-title"
                      >
                        Amala
                      </h4>,
                      <p
                        className="card-text"
                      >
                        This is the decription
                      </p>,
                    ],
                    "data-recipe-card-id": "recipeundefined",
                    "replace": false,
                    "to": "/recipe/undefined",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Amala",
                        "className": "card-title",
                      },
                      "ref": null,
                      "rendered": "Amala",
                      "type": "h4",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "This is the decription",
                        "className": "card-text",
                      },
                      "ref": null,
                      "rendered": "This is the decription",
                      "type": "p",
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "isDownVoted": false,
                    "isFavorited": false,
                    "isUpVoted": false,
                    "recipe": Object {
                      "description": "This is the decription",
                      "downvotes": Array [
                        1,
                        2,
                        4,
                      ],
                      "image": "This is the image",
                      "name": "Amala",
                      "upvotes": Array [
                        1,
                        2,
                        4,
                      ],
                    },
                    "toggleFavouriteRecipe": [Function],
                    "toggleThumbsDownRecipe": [Function],
                    "toggleThumbsUpRecipe": [Function],
                    "user": Object {
                      "firstName": "Seun",
                      "lastName": "Agbeye",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "editRecipe": [Function],
          "id": undefined,
          "recipe": Object {
            "description": "This is the decription",
            "downvotes": Array [
              1,
              2,
              4,
            ],
            "image": "This is the image",
            "name": "Amala",
            "upvotes": Array [
              1,
              2,
              4,
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": undefined,
          "onDelete": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card recipe-card"
          >
            <img
              alt="Card image cap"
              className="card-img-top"
              src="This is the image"
            />
            <div
              className="container"
            >
              <div
                className="card-block"
              >
                <Link
                  data-recipe-card-id="recipeundefined"
                  replace={false}
                  to="/recipe/undefined"
                >
                  <h4
                    className="card-title"
                  >
                    Amala
                  </h4>
                  <p
                    className="card-text"
                  >
                    This is the decription
                  </p>
                </Link>
                <RecipeCardActions
                  isDownVoted={false}
                  isFavorited={false}
                  isUpVoted={false}
                  recipe={
                    Object {
                      "description": "This is the decription",
                      "downvotes": Array [
                        1,
                        2,
                        4,
                      ],
                      "image": "This is the image",
                      "name": "Amala",
                      "upvotes": Array [
                        1,
                        2,
                        4,
                      ],
                    }
                  }
                  toggleFavouriteRecipe={[Function]}
                  toggleThumbsDownRecipe={[Function]}
                  toggleThumbsUpRecipe={[Function]}
                  user={
                    Object {
                      "firstName": "Seun",
                      "lastName": "Agbeye",
                    }
                  }
                />
              </div>
            </div>
          </div>,
          <EditModal
            editRecipe={[Function]}
            id={undefined}
            recipe={
              Object {
                "description": "This is the decription",
                "downvotes": Array [
                  1,
                  2,
                  4,
                ],
                "image": "This is the image",
                "name": "Amala",
                "upvotes": Array [
                  1,
                  2,
                  4,
                ],
              }
            }
          />,
          <DeleteModal
            id={undefined}
            onDelete={[Function]}
          />,
        ],
        "className": "col-xs-12 col-sm-12 col-md-6 col-lg-3 my-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Card image cap"
                className="card-img-top"
                src="This is the image"
              />,
              <div
                className="container"
              >
                <div
                  className="card-block"
                >
                  <Link
                    data-recipe-card-id="recipeundefined"
                    replace={false}
                    to="/recipe/undefined"
                  >
                    <h4
                      className="card-title"
                    >
                      Amala
                    </h4>
                    <p
                      className="card-text"
                    >
                      This is the decription
                    </p>
                  </Link>
                  <RecipeCardActions
                    isDownVoted={false}
                    isFavorited={false}
                    isUpVoted={false}
                    recipe={
                      Object {
                        "description": "This is the decription",
                        "downvotes": Array [
                          1,
                          2,
                          4,
                        ],
                        "image": "This is the image",
                        "name": "Amala",
                        "upvotes": Array [
                          1,
                          2,
                          4,
                        ],
                      }
                    }
                    toggleFavouriteRecipe={[Function]}
                    toggleThumbsDownRecipe={[Function]}
                    toggleThumbsUpRecipe={[Function]}
                    user={
                      Object {
                        "firstName": "Seun",
                        "lastName": "Agbeye",
                      }
                    }
                  />
                </div>
              </div>,
            ],
            "className": "card recipe-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Card image cap",
                "className": "card-img-top",
                "src": "This is the image",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="card-block"
                >
                  <Link
                    data-recipe-card-id="recipeundefined"
                    replace={false}
                    to="/recipe/undefined"
                  >
                    <h4
                      className="card-title"
                    >
                      Amala
                    </h4>
                    <p
                      className="card-text"
                    >
                      This is the decription
                    </p>
                  </Link>
                  <RecipeCardActions
                    isDownVoted={false}
                    isFavorited={false}
                    isUpVoted={false}
                    recipe={
                      Object {
                        "description": "This is the decription",
                        "downvotes": Array [
                          1,
                          2,
                          4,
                        ],
                        "image": "This is the image",
                        "name": "Amala",
                        "upvotes": Array [
                          1,
                          2,
                          4,
                        ],
                      }
                    }
                    toggleFavouriteRecipe={[Function]}
                    toggleThumbsDownRecipe={[Function]}
                    toggleThumbsUpRecipe={[Function]}
                    user={
                      Object {
                        "firstName": "Seun",
                        "lastName": "Agbeye",
                      }
                    }
                  />
                </div>,
                "className": "container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Link
                      data-recipe-card-id="recipeundefined"
                      replace={false}
                      to="/recipe/undefined"
                    >
                      <h4
                        className="card-title"
                      >
                        Amala
                      </h4>
                      <p
                        className="card-text"
                      >
                        This is the decription
                      </p>
                    </Link>,
                    <RecipeCardActions
                      isDownVoted={false}
                      isFavorited={false}
                      isUpVoted={false}
                      recipe={
                        Object {
                          "description": "This is the decription",
                          "downvotes": Array [
                            1,
                            2,
                            4,
                          ],
                          "image": "This is the image",
                          "name": "Amala",
                          "upvotes": Array [
                            1,
                            2,
                            4,
                          ],
                        }
                      }
                      toggleFavouriteRecipe={[Function]}
                      toggleThumbsDownRecipe={[Function]}
                      toggleThumbsUpRecipe={[Function]}
                      user={
                        Object {
                          "firstName": "Seun",
                          "lastName": "Agbeye",
                        }
                      }
                    />,
                  ],
                  "className": "card-block",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <h4
                          className="card-title"
                        >
                          Amala
                        </h4>,
                        <p
                          className="card-text"
                        >
                          This is the decription
                        </p>,
                      ],
                      "data-recipe-card-id": "recipeundefined",
                      "replace": false,
                      "to": "/recipe/undefined",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Amala",
                          "className": "card-title",
                        },
                        "ref": null,
                        "rendered": "Amala",
                        "type": "h4",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "This is the decription",
                          "className": "card-text",
                        },
                        "ref": null,
                        "rendered": "This is the decription",
                        "type": "p",
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "isDownVoted": false,
                      "isFavorited": false,
                      "isUpVoted": false,
                      "recipe": Object {
                        "description": "This is the decription",
                        "downvotes": Array [
                          1,
                          2,
                          4,
                        ],
                        "image": "This is the image",
                        "name": "Amala",
                        "upvotes": Array [
                          1,
                          2,
                          4,
                        ],
                      },
                      "toggleFavouriteRecipe": [Function],
                      "toggleThumbsDownRecipe": [Function],
                      "toggleThumbsUpRecipe": [Function],
                      "user": Object {
                        "firstName": "Seun",
                        "lastName": "Agbeye",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "editRecipe": [Function],
            "id": undefined,
            "recipe": Object {
              "description": "This is the decription",
              "downvotes": Array [
                1,
                2,
                4,
              ],
              "image": "This is the image",
              "name": "Amala",
              "upvotes": Array [
                1,
                2,
                4,
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": undefined,
            "onDelete": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
